### PR TITLE
fix: Fix TradeMarketPriceConversionFeature [skip ci]

### PR DESCRIPTION
**Pull request by ShadowCat117**

### CODE REVIEW

File: common/src/main/java/com/wynntils/features/trademarket/TradeMarketPriceConversionFeature.java [hunk 1]
```diff
@@ -45,13 +44,4 @@ public void onClientChat(ChatSentEvent event) {
...
     }
-
-    @SubscribeEvent
-    public void onGuiOpen(ScreenOpenedEvent.Post event) {
-        if (TRADE_MARKET_PATTERN
-                .matcher(event.getScreen().getTitle().getString())
-                .matches()) {
-            shouldConvert = false;
-        }
-    }
```
Comment (ShadowCat117) [inline common/src/main/java/com/wynntils/features/trademarket/TradeMarketPriceConversionFeature.java:56]: You are no longer able to open the tm until inputting something so I presume this is no longer needed

File: common/src/main/java/com/wynntils/features/trademarket/TradeMarketPriceConversionFeature.java [hunk 2]
```diff
@@ -10,26 +10,25 @@
...
 @ConfigCategory(Category.TRADEMARKET)
 public class TradeMarketPriceConversionFeature extends Feature {
-    private static final Pattern PRICE_PATTERN =
-            Pattern.compile("^§6Type the price in emeralds or type 'cancel' to cancel:$");
-    private static final Pattern TRADE_MARKET_PATTERN = Pattern.compile("^What would you like to sell\\?$");
-    private static final Pattern CANCELLED_PATTERN = Pattern.compile("^You moved and your chat input was canceled\\.$");
+    private static final Pattern PRICE_PATTERN = Pattern.compile(
+            "^§5(\uE00A\uE002|\uE001) \n\uE001 Type the price in emeralds or type \n\uE001 'cancel' to cancel:\n\uE001 ");
+    private static final Pattern CANCELLED_PATTERN =
+            Pattern.compile("^§4(\uE008\uE002|\uE001) You moved and your chat input was canceled.$");
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/features/trademarket/TradeMarketPriceConversionFeature.java:22]: We really should be able to strip these prefixes with RecipientType/StyledTextUtils soon, without adding them to every regex.

**Review by kristofbolyai — APPROVED** [review]

File: common/src/main/java/com/wynntils/features/trademarket/TradeMarketPriceConversionFeature.java (hunk 2 above)
Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/features/trademarket/TradeMarketPriceConversionFeature.java:22]: In fact, if you want to take this up, maybe it'd be the best. Although @DonkeyBlaster also mentioned wanting to work on RecepientTypes. We can not only easily parse the RecipientTypes from the headers, you can also easily get rid of them, as `StyledTextUtils#unwrap` already exists, you'd only need to add a method for removing the initial header.

Comment (ShadowCat117) [reply common/src/main/java/com/wynntils/features/trademarket/TradeMarketPriceConversionFeature.java:22]: Sure I can probably take a look this weekend if it hasn't already been done/started

**Pull request merged**